File: datastructure-problems-java/src/two/pointer/MaximumContainer.java
package two.pointer;

public class MaximumContainer {
    public static int maxArea(int[] heights) {
        int left = 0;
        int right = heights.length - 1;
        int maxArea = 0;
        while (left < right) {
            int height = Math.min((heights[left]), (heights[right]));
            int width = right - left;
            int area = height * width;
            maxArea = Math.max(maxArea, area);
            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxArea;

    }

    public static void main(String[] args) {
        int[] height = {8, 7, 8, 3, 7, 6};
        System.out.println(maxArea(height));
    }
}

File: datastructureproblems-python/two/pointer/palindrome/TwoPointers/MaximumContainer.py
def maxArea(heights):
   max_area = 0
   left=0
   right=len(heights)-1
   while(left<right):
      width=right-left
      min_height=min(heights[left],heights[right])
      max_area=max(min_height*width,max_area)
      if heights[left]<heights[right]:
        left+=1
      else:
        right-=1
   return max_area

if __name__ == "__main__":
   print("Hello World")
   heights = [8, 7, 8, 3, 7, 6]
   print(maxArea(heights))
